modules/virt-runbook-outdatedvirtualmachineinstanceworkloads.adoc

```diff
@@ -0,0 +1,135 @@
+// Module included in the following assemblies:
+//
+// * virt/logging_events_monitoring/virt-runbooks.adoc
+
+:_content-type: REFERENCE
+[id="virt-runbook-outdatedvirtualmachineinstanceworkloads_{context}"]
+= OutdatedVirtualMachineInstanceWorkloads
+
+// Edited by apinnick Nov 2022
+
+[discrete]
+[id="meaning-outdatedvirtualmachineinstanceworkloads_{context}"]
+== Meaning
+
+This alert fires when running virtual machine instances (VMIs) in
+outdated `virt-launcher` pods are detected 24 hours after the OpenShift
+Virtualization control plane has been updated.
+
+[discrete]
+[id="impact-outdatedvirtualmachineinstanceworkloads_{context}"]
+== Impact
+
+Outdated VMIs might not have access to new {VirtProductName}
+features.
+
+Outdated VMIs will not receive the security fixes associated with
+the `virt-launcher` pod update.
+
+[discrete]
+[id="diagnosis-outdatedvirtualmachineinstanceworkloads_{context}"]
+== Diagnosis
+
+. Identify the outdated VMIs:
++
+[source,terminal]
+----
+$ oc get vmi -l kubevirt.io/outdatedLauncherImage --all-namespaces
+----
+
+. Check the `KubeVirt` custom resource (CR) to determine whether
+`workloadUpdateMethods` is configured in the `workloadUpdateStrategy`
+stanza:
++
+[source,terminal]
+----
+$ oc get kubevirt kubevirt --all-namespaces -o yaml
+----
+
+. Check each outdated VMI to determine whether it is live-migratable:
++
+[source,terminal]
+----
+$ oc get vmi <vmi> -o yaml
+----
++
+.Example output
++
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+...
+  status:
+    conditions:
+    - lastProbeTime: null
+      lastTransitionTime: null
+      message: cannot migrate VMI which does not use masquerade
+      to connect to the pod network
+      reason: InterfaceNotLiveMigratable
+      status: "False"
+      type: LiveMigratable
+----
+
+[discrete]
+[id="mitigation-outdatedvirtualmachineinstanceworkloads_{context}"]
+== Mitigation
+
+[discrete]
+[id="configuring-automated-workload-updates-outdatedvirtualmachineinstanceworkloads_{context}"]
+=== Configuring automated workload updates
```

Comment:
Not critical, but this mitigation section looks different than the others. It's not clear to me whether users are supposed to do all three, or just pick one of them.

Reply:
Three different mitigations for three different situations. The user is supposed to pick the one that applies.